lib/ReadOnlyMiddleware.ts

```diff
@@ -0,0 +1,16 @@
+import { Request, Response, NextFunction } from 'express';
+
+const MONGOKU_READ_ONLY_MODE = process.env.MONGOKU_READ_ONLY_MODE === 'true';
+
+export function readOnlyMode(_: Request, res: Response, next: NextFunction) {
+    if (MONGOKU_READ_ONLY_MODE) {
+        res.status(401);
```

Comment:
```suggestion
        // METHOD not allowed
        res.status(405);
```

or `403` (forbidden)